drivers/serial/uart_nrfx_uarte2.c

```diff
@@ -0,0 +1,1053 @@
+	}
+
+#if defined(UARTE_CONFIG_PARITYTYPE_Msk)
+	uarte_cfg.paritytype = NRF_UARTE_PARITYTYPE_EVEN;
+#endif
+	switch (cfg->parity) {
+	case UART_CFG_PARITY_NONE:
+		uarte_cfg.parity = NRF_UARTE_PARITY_EXCLUDED;
+		break;
+	case UART_CFG_PARITY_EVEN:
+		uarte_cfg.parity = NRF_UARTE_PARITY_INCLUDED;
+		break;
+#if defined(UARTE_CONFIG_PARITYTYPE_Msk)
+	case UART_CFG_PARITY_ODD:
+		uarte_cfg.parity = NRF_UARTE_PARITY_INCLUDED;
+		uarte_cfg.paritytype = NRF_UARTE_PARITYTYPE_ODD;
+		break;
+#endif
+	default:
+		return -ENOTSUP;
+	}
+
+	if (baudrate_set(nrfx_dev->p_reg, cfg->baudrate) != 0) {
+		return -ENOTSUP;
+	}
+
+	nrfy_uarte_configure(nrfx_dev->p_reg, &uarte_cfg);
+
+	data->uart_config = *cfg;
+
+	return 0;
+}
+
+static int uarte_nrfx_config_get(const struct device *dev,
+				 struct uart_config *cfg)
+{
+	struct uarte_nrfx_data *data = dev->data;
+
+	*cfg = data->uart_config;
+	return 0;
+}
+#endif /* CONFIG_UART_USE_RUNTIME_CONFIGURE */
+
+#if UARTE_ANY_POLLING || UARTE_ANY_INTERRUPT_DRIVEN
+static int api_err_check(const struct device *dev)
+{
+	if (is_sync_api(dev)) {
+		const struct uarte_nrfx_config *cfg = dev->config;
+		const nrfx_uarte_t *instance = &cfg->nrfx_dev;
+		uint32_t mask = nrfx_uarte_errorsrc_get(instance);
+
+		return mask;
+	}
+
+	struct uarte_nrfx_data *data = dev->data;
+	uint32_t rv = data->async->err;
+
+	data->async->err = 0;
+
+	return rv;
+}
+#endif
+
+static const struct uart_async_to_irq_async_api a2i_api = {
+	.callback_set		= callback_set,
+	.tx			= api_tx,
+	.tx_abort		= api_tx_abort,
+	.rx_enable		= api_rx_enable,
+	.rx_buf_rsp		= api_rx_buf_rsp,
+	.rx_disable		= api_rx_disable,
+};
+
+static const struct uart_driver_api uart_nrfx_uarte_driver_api = {
+	.poll_in	= api_poll_in,
+	.poll_out	= api_poll_out,
+#ifdef CONFIG_UART_USE_RUNTIME_CONFIGURE
+	.configure	= uarte_nrfx_configure,
+	.config_get	= uarte_nrfx_config_get,
+#endif /* CONFIG_UART_USE_RUNTIME_CONFIGURE */
+#if UARTE_ANY_POLLING || UARTE_ANY_INTERRUPT_DRIVEN
+	.err_check	= api_err_check,
+#endif
+#if UARTE_ANY_ASYNC
+	.callback_set	= api_callback_set,
+	.tx		= api_tx,
+	.tx_abort	= api_tx_abort,
+	.rx_enable	= api_rx_enable,
+	.rx_buf_rsp	= api_rx_buf_rsp,
+	.rx_disable	= api_rx_disable,
+#endif /* UARTE_ANY_ASYNC */
+#if UARTE_ANY_INTERRUPT_DRIVEN
+	UART_ASYNC_TO_IRQ_API_INIT(),
+#endif /* UARTE_ANY_INTERRUPT_DRIVEN */
+};
+
+static int endtx_stoptx_ppi_init(NRF_UARTE_Type *uarte)
+{
+	nrfx_err_t ret;
+	uint8_t ch;
+
+	ret = nrfx_gppi_channel_alloc(&ch);
+	if (ret != NRFX_SUCCESS) {
+		LOG_ERR("Failed to allocate PPI Channel");
+		return -EIO;
+	}
+
+	nrfx_gppi_channel_endpoints_setup(ch,
+		nrfy_uarte_event_address_get(uarte, NRF_UARTE_EVENT_ENDTX),
+		nrfy_uarte_task_address_get(uarte, NRF_UARTE_TASK_STOPTX));
+	nrfx_gppi_channels_enable(BIT(ch));
+
+	return 0;
+}
+
+static int start_rx(const struct device *dev)
+{
+	const struct uarte_nrfx_config *cfg = dev->config;
+
+	if (cfg->flags & UARTE_CFG_FLAG_NO_RX) {
+		return -ENOTSUP;
```

Comment:
`start_rx()` is called at initialization of the device, so this will cause the initialization to fail for instances not using the async API but having the `disable-rx` property set.

Reply:
Added error code conversion from -ENOTSUP to 0 in the init function so that -ENOTSUP is also accepted.